### ISSUE TITLE
non-literal floats capture all in match statements 

### ISSUE THREAD

**Issue by Jeremiah-England** [microsoft/pyright#4899]
**Describe the bug**

I am not completely sure this is a bug but it seems like one to me. If a float is not typed as a `Literal`, then it captures all the floats in a `match` statement. See the example below. My guess is this was done so `float()` catches all the floats? But that seems like a special case that should be handled instead of assuming everything that has the type `float` will catch all the cases.

**To Reproduce**

I am on the pyright version 1.1.301.

```python
from typing_extensions import reveal_type


class SpecialNumbers:
    SUPRISE_NUMBER = 123
    INFINITY = float("inf")  # I cannot find a way to represent an infinity with Literal...
    ZERO = float()  # This type could be infered as `Literal[0.0]`. But it's `float` instead.


def parse_float(s: str):
    match float(s):
        case SpecialNumbers.SUPRISE_NUMBER:
            return "Hahaha"
        case SpecialNumbers.INFINITY:  # If you comment this case out you'll see the "Never" become "float" below.
            raise NotImplementedError()
        # case SpecialNumbers.ZERO:  # This is another case that behaves like the INFINITY above.
        #     return "zero"
        case f:
            reveal_type(f)  # Type of "f" is "Never"
            return f

# Some useful checks for verify understanding of the actual behavior...
print(parse_float("100"))
print(parse_float("0"))
print(parse_float("123"))
```

**Expected behavior**

I expect the only cases that match all floats to be `float()` or a new variable declared in the match case (`case my_new_float:`). Everything else such as `math.inf` should pass the rest of the floats on to the next case.

Thanks for much for working in Pyright! It's awesome.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.302, which I just published. It will also be included in a future release of pylance.
